operator/cilium_node.go

```diff
@@ -1,112 +1,486 @@
-						if oldNode.DeepEqual(node) {
+					if newNode := k8s.ObjToCiliumNode(newObj); newNode != nil {
+						if oldNode.DeepEqual(newNode) {
+							return
+						}
+						key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(newObj)
+						if err != nil {
+							log.WithError(err).Warning("Unable to process CiliumNode Update event")
 							return
 						}
-						// node is deep copied before it is stored in pkg/aws/eni
-						nodeManager.Update(node)
+						cfg.queue.Add(key)
+					} else {
+						log.Warningf("Unknown CiliumNode object type %T received: %+v", newNode, newNode)
 					}
+				} else {
+					log.Warningf("Unknown CiliumNode object type %T received: %+v", oldNode, oldNode)
 				}
 			},
 			DeleteFunc: func(obj interface{}) {
-				if node := k8s.ObjToCiliumNode(obj); node != nil {
-					nodeManager.Delete(node.Name)
+				key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+				if err != nil {
+					log.WithError(err).Warning("Unable to process CiliumNode Delete event")
+					return
 				}
+				cfg.queue.Add(key)
 			},
-		},
-		k8s.ConvertToCiliumNode,
+		}
+	} else {
+		// Since we won't be handling any events we don't need to convert
+		// objects.
+		cfg.ciliumNodeConvertFunc = nil
+	}
+
+	return cfg, nil
+}
+
+// ciliumNodeName is only used to implement NamedKey interface.
+type ciliumNodeName struct {
+	name string
+}
+
+func (c *ciliumNodeName) GetKeyName() string {
+	return c.name
+}
+
+func startSynchronizingCiliumNodes(cfg *ciliumNodeWatcherCfg) error {
+	if cfg.withKVStore {
+		log.Info("Starting to synchronize CiliumNode custom resources to KVStore")
+	} else {
+		log.Info("Starting to synchronize CiliumNode custom resources")
+	}
+
+	// TODO: The operator is currently storing a full copy of the
+	// CiliumNode resource, as the resource grows, we may want to consider
+	// introducing a slim version of it.
+	var ciliumNodeInformer cache.Controller
+	ciliumNodeStore, ciliumNodeInformer = informer.NewInformer(
+		cache.NewListWatchFromClient(ciliumK8sClient.CiliumV2().RESTClient(),
+			cilium_v2.CNPluralName, v1.NamespaceAll, fields.Everything()),
+		&cilium_v2.CiliumNode{},
+		0,
+		cfg.resourceEventHandler,
+		cfg.ciliumNodeConvertFunc,
 	)
 
 	go func() {
 		cache.WaitForCacheSync(wait.NeverStop, ciliumNodeInformer.HasSynced)
 		close(k8sCiliumNodesCacheSynced)
+		// Only handle events if syncHandler is not nil. If it is nil then
+		// there isn't any event handler set for CiliumNodes events.
+		if cfg.syncHandler != nil {
+			for processNextWorkItem(cfg.queue, cfg.syncHandler) {
+			}
+		}
 	}()
 
 	go ciliumNodeInformer.Run(wait.NeverStop)
+
+	return nil
 }
 
-func deleteCiliumNode(nodeManager *allocator.NodeEventHandler, name string) {
-	if nodeManager != nil {
-		(*nodeManager).Delete(name)
+// processNextWorkItem process all events from the workqueue. See
+// workqueue.RateLimitingInterface documentation to understand why certain
+// methods are executed in certain conditions.
```

Comment:
What does this comment mean? I looked up https://pkg.go.dev/k8s.io/client-go/util/workqueue#RateLimitingInterface and I'm now more confused which conditions or methods this comment is trying to explain.